test/property.test.js

```diff
@@ -51,36 +75,215 @@ describe('loopback:property generator', function() {
       expect(props).to.have.property('isPreferred');
       expect(props.isPreferred).to.eql({
         type: 'boolean',
-        required: true,
         default: true
       });
       done();
     });
   });
 
+  it('creates date type property from ISO string', function(done) {
+    var propertyGenerator = givenPropertyGenerator();
+    helpers.mockPrompt(propertyGenerator, {
+      model: 'Car',
+      name: 'year',
+      type: 'date',
+      defaultValue: '2015-11'
+    });
+
+    propertyGenerator.run(function() {
+      var definition = common.readJsonSync('common/models/car.json');
+      var props = definition.properties || {};
+      expect(props).to.have.property('year');
+      expect(props.year).to.eql({
+        type: 'date',
+        default: '2015-11-01T00:00:00.000Z'
+      });
+      done();
+    });
+  });
+
+  it('creates date type property from number', function(done) {
+    var propertyGenerator = givenPropertyGenerator();
+    helpers.mockPrompt(propertyGenerator, {
+      model: 'Car',
+      name: 'year',
+      type: 'date',
+      defaultValue: '1466087191000'
+    });
+
+    propertyGenerator.run(function() {
+      var definition = common.readJsonSync('common/models/car.json');
+      var props = definition.properties || {};
+      expect(props).to.have.property('year');
+      expect(props.year).to.eql({
+        type: 'date',
+        default: '2016-06-16T14:26:31.000Z'
+      });
+      done();
+    });
+  });
+
   it('creates a typed array', function(done) {
     var propertyGenerator = givenPropertyGenerator();
     helpers.mockPrompt(propertyGenerator, {
       model: 'Car',
       name: 'list',
       type: 'array',
-      itemType: 'string'
+      customType: '', // temporary workaround for
+                      // https://github.com/yeoman/generator/issues/600
+      customItemType: '', // temporary workaround for
+                          // https://github.com/yeoman/generator/issues/600
+      itemType: 'object'
```

Comment:
Ah, I put the comment on a wrong line, sorry for that. I meant lines 132-135, `customType` and `customItemType`.

This line 136, containing `itemType` property, is ok.